packages/node/src/indexer/mmr.service.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import fs from 'fs';
+import { Injectable, OnApplicationShutdown } from '@nestjs/common';
+import { u8aToHex, u8aEq } from '@polkadot/util';
+import { MMR } from 'merkle-mountain-range';
+import { Sequelize, Op } from 'sequelize';
+import { NodeConfig } from '../configure/NodeConfig';
+import { SubqueryProject } from '../configure/project.model';
+import { getLogger } from '../utils/logger';
+import { delay } from '../utils/promise';
+import { FileBasedDb, keccak256FlyHash } from '../vendor/merkle-mountain-range';
+import { MetadataFactory, MetadataRepo } from './entities/Metadata.entity';
+import { PoiFactory, PoiRepo, ProofOfIndex } from './entities/Poi.entity';
+
+const logger = getLogger('mmr');
+const DEFAULT_WORD_SIZE = 32;
+const DEFAULT_LEAF = Buffer.from('0x000000000000000000000000000000');
+const DEFAULT_FETCH_RANGE = 100;
+
+@Injectable()
+export class MmrService implements OnApplicationShutdown {
+  private isShutdown = false;
+  private blockOffset: number;
+  private metadataRepo: MetadataRepo;
+  private fileBasedMmr: MMR;
+  private poiRepo: PoiRepo;
+  private mmrStartHeight: number;
```

Comment:
can u use a better name, because `mmrStartHeight` doesn't sound like it will be changed.